src/test/java/org/prebid/server/auction/PrivacyEnforcementServiceTest.java

```diff
@@ -0,0 +1,615 @@
+                singletonMap(BIDDER_NAME, 1)),
+                bidRequestBuilder -> bidRequestBuilder
+                        .user(user)
+                        .device(device)
+                        .regs(regs));
+
+        // when
+        final Future<Map<String, PrivacyEnforcementResult>> masked = privacyEnforcementService
+                .mask(bidderToUser, null, extRegs, singletonList(BIDDER_NAME), emptyMap(), bidRequest, PUBLISHER_ID, timeout);
+
+        // then
+        final PrivacyEnforcementResult expected = PrivacyEnforcementResult.of(user, device, regs);
+
+        verify(gdprService).isGdprEnforced(eq("1"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout));
+        verifyNoMoreInteractions(gdprService);
+        assertThat(masked.succeeded()).isTrue();
+        assertThat(masked.result()).hasSize(1)
+                .containsOnly(entry(BIDDER_NAME, expected));
+    }
+
+    @Test
+    public void shouldNotMaskWhenDeviceLmtIsZeroAndRegsCoppaIsZeroAndExtRegsGdprIsZeroGdprServiceRespondIsGdprEnforcedWithFalse() {
+        // given
+        given(gdprService.isGdprEnforced(eq("0"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout)))
+                .willReturn(Future.succeededFuture(false));
+
+        final ExtRegs extRegs = ExtRegs.of(0);
+        final Regs regs = Regs.of(0, mapper.valueToTree(extRegs));
+        final User user = notMaskedUser();
+        final Device device = givenNotMaskedDevice(deviceBuilder -> deviceBuilder.lmt(0));
+        final Map<String, User> bidderToUser = singletonMap(BIDDER_NAME, user);
+
+        final BidRequest bidRequest = givenBidRequest(givenSingleImp(
+                singletonMap(BIDDER_NAME, 1)),
+                bidRequestBuilder -> bidRequestBuilder
+                        .user(user)
+                        .device(device)
+                        .regs(regs));
+
+        // when
+        final Future<Map<String, PrivacyEnforcementResult>> masked = privacyEnforcementService
+                .mask(bidderToUser, null, extRegs, singletonList(BIDDER_NAME), emptyMap(), bidRequest, PUBLISHER_ID, timeout);
+
+        // then
+        final PrivacyEnforcementResult expected = PrivacyEnforcementResult.of(user, device, regs);
+
+        verify(gdprService).isGdprEnforced(eq("0"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout));
+        verifyNoMoreInteractions(gdprService);
+
+        assertThat(masked.succeeded()).isTrue();
+        assertThat(masked.result()).hasSize(1)
+                .containsOnly(entry(BIDDER_NAME, expected));
+    }
+
+    @Test
+    public void shouldResolveBidderNameByAliases() {
+        // given
+        given(gdprService.isGdprEnforced(eq("1"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout)))
+                .willReturn(Future.succeededFuture(true));
+        given(gdprService.resultByVendor(eq(singleton(15)), eq("1"), any(), any(), eq(timeout)))
+                .willReturn(Future.succeededFuture(GdprResponse.of(true, singletonMap(15, false), null)));
```

Comment:
Same as for `gdprService.isGdprEnforced` stub - check what `willReturn` is the most common one and declare specific ones where needed.